src/methods/scgpt_fine_tuned/script.py

```diff
@@ -0,0 +1,463 @@
+import sys
+import tarfile
+import tempfile
+import time
+import zipfile
+import anndata as ad
+import json
+import os
+import gdown
+import torch
+import numpy as np
+from scipy.sparse import issparse
+import torch
+from torchtext.vocab import Vocab
+import scgpt
+from sklearn.model_selection import train_test_split
+
+
+## VIASH START
+par = {
+  'input_train': 'resources_test/task_label_projection/cxg_immune_cell_atlas/train.h5ad',
+  'input_test': 'resources_test/task_label_projection/cxg_immune_cell_atlas/test.h5ad',
+  'output': 'output.h5ad',
+  'model_name': 'scGPT_human',
+  'model': 'scGPT_human'
+}
+meta = {
+  'name': 'scgpt_fine_tuned',
+  'temp_dir': 'tmp'
+}
+## VIASH END
+
+
+sys.path.append(meta["resources_dir"])
+from functions import prepare_data, prepare_dataloader, train, test, evaluate
+
+
+### Load input data and model
+print("Reading input files", flush=True)
+input_train = ad.read_h5ad(par["input_train"])
+input_test = ad.read_h5ad(par["input_test"])
+
+if input_train.uns["dataset_organism"] != "homo_sapiens":
+  raise ValueError(
+    f"scGPT can only be used with human data "
+    f"(dataset_organism is \"{input_train.uns['dataset_organism']}\")"
+  )
+
+if par["model"] is None:
+  print(f"\n>>> Downloading '{par['model_name']}' model...", flush=True)
+  model_drive_ids = {
+    "scGPT_human": "1oWh_-ZRdhtoGQ2Fw24HP41FgLoomVo-y",
+    "scGPT_CP": "1_GROJTzXiAV8HB4imruOTk6PEGuNOcgB",
```

Comment:
I don't think the CP model is meant to be fine tuned so it can probably be removed (not that it will be used anyway).

Reply:
Oh yes, I'll remove it. Thanks for noticing!